File: 20260519/exos/05_defaut.py
# ============================================================
# Hearts of Iron IV — mobilisation
# ============================================================
# Le général peut appeler des divisions. Sans précision, le jeu
# en mobilise 10 par défaut. Avec un ordre urgent, tu peux en demander plus.
#
# Lance : python3 05_defaut.py
# ============================================================

#   n  : nombre de divisions demandées (optionnel)
#
# mobiliser()           →  pas d'argument  →  retourne 10
# mobiliser(25)         →  n vaut 25       →  retourne 25
#
# Résultat attendu : mobiliser() == 10  et  mobiliser(25) == 25
#
# Indice : def mobiliser(n=10): puis return n

def mobiliser(n=10):
    return n



# --- Vérification (ne pas modifier) ---
assert mobiliser() == 10, "mobiliser() sans argument doit retourner 10"
assert mobiliser(25) == 25, "mobiliser(25) doit retourner 25"
print("✅ Correct !")

File: 20260724/exos/05_defaut.lua
-- ============================================================
-- Hearts of Iron IV — nom de division par défaut
-- ============================================================
-- Si le joueur ne donne pas de nom (nil), on utilise "Division".
--
-- Lance : lua 05_defaut.lua
-- ============================================================

-- nommer("Panzer")  →  "Panzer"
-- nommer(nil)       →  "Division"
--
-- Indice : nom = nom or "Division"  puis  return nom

function nommer(nom)
    nom = nom or "Division"
    return nom
end


-- --- Vérification (ne pas modifier) ---
assert(nommer("Panzer") == "Panzer", "nommer('Panzer') doit retourner 'Panzer'")
assert(nommer(nil) == "Division", "nommer(nil) doit retourner 'Division'")
print("✅ Correct !")
